jax/_src/mesh_utils.py

```diff
@@ -597,10 +597,10 @@ def _generate_logical_mesh(
           zip(logical_indices, physical_indices, range(len(logical_indices)))
       )
   )
-  logical_mesh = np.transpose(logical_mesh, transpose_axes)  # type: ignore  # numpy 2.2
+  logical_mesh = np.transpose(logical_mesh, transpose_axes)  # numpy 2.2
```

Comment:
There are these # numpy 2.2 comments lying around in several files. 

Run your magic incantation on these comments too?